src/envoy/http/service_control/client_cache.cc

```diff
@@ -381,37 +388,50 @@ void ClientCache::handleCheckResponse(const Status& http_status,
     }
   } else {
     if (translate_non_5xx) {
```

Comment:
translate_non_5xx is kind of confusing for me.   how about final_status_is_from_http? is this correct?

Reply:
Correct. We only want to translate HTTP 4xx errors, not SC Check 4xx errors. I think I can simplify this actually, no need for the variable.